pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -795,6 +798,10 @@ allChainsLabel:
 			filterChainMatch = &fcm
 			if filterChainMatchOption[id].Protocol == istionetworking.ListenerProtocolHTTP {
 				httpOpts = configgen.buildSidecarInboundHTTPListenerOptsForPortOrUDS(node, pluginParams)
+				if chain.TLSContext != nil && chain.TLSContext.CommonTlsContext != nil {
```

Comment:
FYI, IIUC, tls_context is being deprecated by `transport_socket`,  I think we finished that in the pilot code already, could you confirm that? if so, you should check that field instead. @howardjohn for extra coverage.

https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/listener/listener_components.proto#envoy-api-msg-listener-filterchain

Reply:
I think this tlscontext points to DownstreamTlsContext that is part of the transport_socket that is built for the listener in buildDownstreamTLSTransportSocket function...